activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -60,6 +60,17 @@ def locking_enabled? #:nodoc:
         self.class.locking_enabled?
       end
 
+      def increment!(attribute, by = 1, touch: nil) #:nodoc:
+        super
+
+        if locking_enabled?
+          self[self.class.locking_column] += 1
+          clear_attribute_change(self.class.locking_column)
+        end
+
+        self
+      end
```

Comment:
I'd prefer to avoid duplicating the method signature and return value:

```suggestion
      def increment!(*, **) #:nodoc:
        super.tap do
          if locking_enabled?
            self[self.class.locking_column] += 1
            clear_attribute_change(self.class.locking_column)
          end
        end
      end
```

Reply:
Thanks, rebased accordingly. (I was unfamiliar with that pattern, that's very handy.)